### ISSUE TITLE
add :unique property for def-vertex and def-edge

### ISSUE THREAD

**Issue by kraison** [kraison/vivace-graph-v3#6]
Unique slot values would be very useful.  Enforcing uniqueness now is cumbersome.

**Comment by fiddlerwoaroof** [comment]
Is this implemented?

https://github.com/kraison/vivace-graph-v3/blob/b376c4372adc74126ec42c79f705421b977e8c6a/srn.lisp#L99

**Comment by kraison** [comment]
No, it is not implemented;  that reference is from VG2.  I have removed it to avoid confusion.

**Comment by kraison** [comment]
Acknowledged as a planned feature. A `:unique` slot option would be enforced by a unique secondary index — a duplicates-disallowed index keyed by the slot value, built on the existing view/skip-list machinery, with the insert/update path signalling a duplicate-key error on violation. Deferring implementation for now. Tracking.

**Comment by kraison** [comment]
And, if we are going to do this, we might consider just adding an :indexed option for slots generally. If we have the plumbing for a unique index, a non-unique index is trivial.

**Comment by kraison** [comment]
Implemented on the `experiment` branch. `:unique` is now a first-class slot property on `def-vertex` / `def-edge`.

- **Declaration:** `(slot :type … :unique t | equal | equalp | <canonicalizer>)` — the value is the uniqueness key (identity, case/edge folding, or an arbitrary canonicalizer like `string-downcase`).
- **Semantics:** enforced at the commit boundary — a violation aborts the whole transaction with `unique-constraint-violation`; NULL-exempt (SQL-style); shared across subclasses of the declaring type; concurrent commits racing for the same value are serialized so exactly one wins.
- **Durable:** backed by a persistent unique index reopened with the graph (skip-list *or* B+ tree backend; in-RAM map on a memory-graph) — not rebuilt by scanning every node on open.
- **Tests:** `tests/unique-constraint-tests.lisp` + `tests/peer-unique-tests.lisp`, green on SBCL/ECL.
- **Docs:** manual §8, "Unique constraints (`:unique`)".

The remaining distributed corner — cross-device collisions under global scope, where a hub must arbitrate — is tracked in #51. Closing, as the single-graph feature is complete.
